src/extension/tiltify-client.ts

```diff
@@ -0,0 +1,132 @@
+import axios from 'axios';
+import { get } from '../extension/util/nodecg';
+import { tiltifyTokens } from '../extension/util/replicants';
+import { Configschema } from '../types/generated/configschema';
+import {
+  TiltifyTokenResponse,
+  TiltifyCampaignData,
+  TiltifyRawDonation,
+  TiltifyMilestone,
+  TiltifyDonation
+} from '../types/custom/Tiltify-Types';
+
+const nodecg = get();
+const config = (nodecg.bundleConfig as Configschema).tiltify;
+
+export class TiltifyClient {
+  private readonly clientId: string;
+  private readonly clientSecret: string;
+  public readonly campaignId: string;
+  private readonly TOKEN_EXPIRY_BUFFER = 120;
+
+  constructor() {
+    if (!config?.enable) throw new Error('Tiltify integration disabled');
+    if (!config.clientId || !config.clientSecret || !config.campaign) {
+      throw new Error('Missing Tiltify configuration values');
+    }
```

Comment:
```suggestion
    if (config.enabled && (!config.clientId || !config.clientSecret || !config.campaign)) {
      throw new Error('Missing Tiltify configuration values');
    }
```

Throwing an error if the integration is disabled doesn't seem very useful? Especially considering it shouldn't even get to such a state because of the checks in `tiltify.ts`. Throwing an error if it's enabled, but the required config variables aren't set is enough IMO.